src/Microsoft.VisualStudio.VsInteractiveWindow/IVsInteractiveWindowFactory2.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.InteractiveWindow.Shell
+{
+    public interface IVsInteractiveWindowFactory2: IVsInteractiveWindowFactory 
+    {
+        IVsInteractiveWindow Create(Guid providerId, int instanceId, string title, IInteractiveEvaluator evaluator, IVsInteractiveWindowDecorator decorator, __VSCREATETOOLWIN creationFlags = 0);
```

Comment:
I'm not sure if there is much value in avoiding additional interface. It'd be if this was version 1. Since we already need Factory2, then having to add Factory3, etc. isn't much worse.

---
In reply to: [90067669](https://github.com/dotnet/interactive-window/pull/23#discussion_r90067669) [](ancestors = 90067669)